src/cmd/src/standalone.rs

```diff
@@ -196,74 +215,37 @@ struct StartCommand {
 }
 
 impl StartCommand {
-    async fn build(self) -> Result<Instance> {
+    fn load_options(&self, top_level_options: TopLevelOptions) -> Result<ConfigOptions> {
         let enable_memory_catalog = self.enable_memory_catalog;
         let config_file = self.config_file.clone();
-        let plugins = Arc::new(load_frontend_plugins(&self.user_provider)?);
-        let fe_opts = FrontendOptions::try_from(self)?;
-        let dn_opts: DatanodeOptions = {
-            let mut opts: StandaloneOptions = if let Some(path) = config_file {
-                toml_loader::from_file!(&path)?
-            } else {
-                StandaloneOptions::default()
-            };
-            opts.enable_memory_catalog = enable_memory_catalog;
-            opts.datanode_options()
-        };
-
-        info!(
-            "Standalone frontend options: {:#?}, datanode options: {:#?}",
-            fe_opts, dn_opts
-        );
-
-        let datanode = Datanode::new(dn_opts.clone())
-            .await
-            .context(StartDatanodeSnafu)?;
-
-        let mut frontend = build_frontend(plugins.clone(), datanode.get_instance()).await?;
-
-        frontend
-            .build_servers(&fe_opts)
-            .await
-            .context(StartFrontendSnafu)?;
-
-        Ok(Instance { datanode, frontend })
-    }
-}
-
-/// Build frontend instance in standalone mode
-async fn build_frontend(
-    plugins: Arc<Plugins>,
-    datanode_instance: InstanceRef,
-) -> Result<FeInstance> {
-    let mut frontend_instance = FeInstance::try_new_standalone(datanode_instance.clone())
-        .await
-        .context(StartFrontendSnafu)?;
-    frontend_instance.set_plugins(plugins.clone());
-    Ok(frontend_instance)
-}
-
-impl TryFrom<StartCommand> for FrontendOptions {
-    type Error = Error;
-
-    fn try_from(cmd: StartCommand) -> std::result::Result<Self, Self::Error> {
-        let opts: StandaloneOptions = if let Some(path) = cmd.config_file {
+        let mut opts: StandaloneOptions = if let Some(path) = config_file {
```

Comment:
The clone above is unnecessary
```suggestion
        let mut opts: StandaloneOptions = if let Some(path) = &self.config_file {
```